packages/pyolite-kernel/py/pyolite/pyolite/interpreter.py

```diff
@@ -19,9 +29,8 @@ def display(self, result):
         return
 
     async def run(self, code):
-        self.run_complete = ensure_future(
-            self.load_packages_and_run(self.run_complete, code)
-        )
-        result = await self.run_complete
-        if result is not None:
-            display(result)
+        exec_code = self.transform_cell(code)
+        await _load_packages_from_imports(exec_code)
+        self.result = self.run_cell(code)
+        if self.result is not None:
+            display(self.result)
```

Comment:
We'll have to remove this `display(self.result)` and use a custom displayhook class instead

See
https://github.com/ipython/ipython/blob/e37bc7afddaf74327eabf661b4ebd4e1047299b5/IPython/core/interactiveshell.py#L444

Reply:
aye aye captain 🚀